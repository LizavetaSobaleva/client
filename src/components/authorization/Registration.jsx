import React, { useState } from "react";
import { Link } from 'react-router-dom';
import "./authorization.less";
import Input from "../UI/input/Input";
import PrimaryButton from "../UI/primaryButton/PrimaryButton";
import { useDispatch } from 'react-redux';
import { registration } from "../../actions/user";

const Registration = () => {
    const [name, setName] = useState("")
    const [email, setEmail] = useState("")
    const [password, setPassword] = useState("")
    const dispatch = useDispatch()


  return (
    <div className="authorization">
      <div className="authorization__container">
        <div className="authorization__label">
          <h1>Create your account</h1>
        </div>
        <div className="authorization__form">
          <div className="authorization__item">
            <label>Name</label>
            <Input value={name} onChange={(event) => setName(event.target.value)} type="text" placeholder="Enter your full name" />
          </div>

          <div className="authorization__item">
            <label>Email</label>
            <Input value={email} onChange={(event) => setEmail(event.target.value)} type="email" placeholder="Enter your email" />
          </div>

          <div className="authorization__item">
            <label>Password</label>
            <Input value={password} onChange={(event) => setPassword(event.target.value)} type="password" placeholder="Enter your password" />
          </div>

          <PrimaryButton onClick={() => registration(name, email, password, dispatch)}>Create Account</PrimaryButton>
        </div>
        <div className="authorization__login">
          Already have an account? <Link to="/login">Log in</Link>
        </div>
      </div>
    </div>
  );
};

export default Registration;
