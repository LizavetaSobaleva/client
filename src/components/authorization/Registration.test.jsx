import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import Registration from './Registration';

const mockStore = configureStore([]);
const store = mockStore({});

jest.mock('../../actions/user', () => ({
  registration: jest.fn(),
}));

describe('Registration Component', () => {
  beforeEach(() => {
    store.dispatch = jest.fn();
  });

  const renderComponent = () => {
    return render(
      <Provider store={store}>
        <MemoryRouter>
          <Registration />
        </MemoryRouter>
      </Provider>
    );
  };

  test('renders Registration form without crashing', () => {
    const { asFragment } = renderComponent();
    expect(screen.getByTestId('formTitle')).toBeInTheDocument();
    expect(screen.getByTestId('nameInput')).toBeInTheDocument();
    expect(screen.getByTestId('emailInput')).toBeInTheDocument();
    expect(screen.getByTestId('passwordInput')).toBeInTheDocument();
    expect(screen.getByTestId('registerButton')).toBeInTheDocument();
    expect(screen.getByTestId('loginLink')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot(); // Add snapshot test
  });

  test('allows user to input name', () => {
    renderComponent();
    const nameInput = screen.getByTestId('nameInput');
    fireEvent.change(nameInput, { target: { value: 'John Doe' } });
    expect(nameInput).toHaveValue('John Doe');
  });

  test('allows user to input email', () => {
    renderComponent();
    const emailInput = screen.getByTestId('emailInput');
    fireEvent.change(emailInput, { target: { value: 'john.doe@example.com' } });
    expect(emailInput).toHaveValue('john.doe@example.com');
  });

  test('allows user to input password', () => {
    renderComponent();
    const passwordInput = screen.getByTestId('passwordInput');
    fireEvent.change(passwordInput, { target: { value: 'password123' } });
    expect(passwordInput).toHaveValue('password123');
  });

  test('register button triggers registration action', () => {
    const { registration } = require('../../actions/user');

    renderComponent();
    fireEvent.change(screen.getByTestId('nameInput'), { target: { value: 'John Doe' } });
    fireEvent.change(screen.getByTestId('emailInput'), { target: { value: 'john.doe@example.com' } });
    fireEvent.change(screen.getByTestId('passwordInput'), { target: { value: 'password123' } });

    fireEvent.click(screen.getByTestId('registerButton'));

    expect(registration).toHaveBeenCalledWith('John Doe', 'john.doe@example.com', 'password123', expect.any(Function));
  });
});
