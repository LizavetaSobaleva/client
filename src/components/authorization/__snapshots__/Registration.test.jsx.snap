// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Component renders Registration form without crashing 1`] = `
<DocumentFragment>
  <div
    class="authorization"
  >
    <div
      class="authorization__container"
    >
      <div
        class="authorization__label"
        data-testid="formTitle"
      >
        <h1>
          Create your account
        </h1>
      </div>
      <div
        class="authorization__form"
      >
        <div
          class="authorization__item"
        >
          <label
            for="nameInput"
          >
            Name
          </label>
          <input
            data-testid="nameInput"
            id="nameInput"
            placeholder="Enter your full name"
            type="text"
            value=""
          />
        </div>
        <div
          class="authorization__item"
        >
          <label
            for="emailInput"
          >
            Email
          </label>
          <input
            data-testid="emailInput"
            id="emailInput"
            placeholder="Enter your email"
            type="email"
            value=""
          />
        </div>
        <div
          class="authorization__item"
        >
          <label
            for="passwordInput"
          >
            Password
          </label>
          <input
            data-testid="passwordInput"
            id="passwordInput"
            placeholder="Enter your password"
            type="password"
            value=""
          />
        </div>
        <button
          class="primaryButton"
          data-testid="registerButton"
        >
          Create Account
        </button>
      </div>
      <div
        class="authorization__login"
        data-testid="loginLink"
      >
        Already have an account? 
        <a
          href="/login"
        >
          Log in
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
